featurebyte/migration/service/mixin.py

```diff
@@ -183,19 +143,7 @@ async def get_session(self, feature_store: FeatureStoreModel) -> BaseSession:
         -------
         BaseSession
         """
-        credential_provider = MongoBackedCredentialProvider(persistent=self.persistent)
-        user = User(id=feature_store.user_id)
```

Comment:
I think it is important that the `user` here is the feature store's user since the session services below need to be specific to that user. See this PR: https://github.com/featurebyte/featurebyte/pull/681

If I recall correctly, the user id that gets passed when calling `run_migration()` is a dummy value.

@gee-senbong could you confirm?

Reply:
YS is right. We need to use the feature store's `user_id` to construct the session. Otherwise, it will cause issue when connecting to the warehouse during migration.

Reply:
ah ok! let me fix this

Reply:
just to double check - this is the entry point where the user id being passed in is the dummy one yea @gee-senbong / @YS-L 

https://github.com/featurebyte/featurebyte/blob/27032840d56d21add4e597b3b1569588316d46f5/docker/migration.py#L19-L21

Reply:
Yes, that's right. There is a similar entry point in the featurebyte-app as well.

Reply:
hmm.. ok, i'm going to have to sit on this a bit more.

Reply:
curious - why do we even pass in a user here into `run_migration` if it's going to be updated at at later point?

i see that we initialize a `SchemaMetadataService` with the arbitrary user id - i guess that's ok?
https://github.com/featurebyte/featurebyte/blob/27032840d56d21add4e597b3b1569588316d46f5/featurebyte/migration/run.py#L210-L215

Reply:
i added some special handling for these services here! @YS-L / @gee-senbong